File: fe/src/api/system/role.ts
import request from '@/utils/request';
import { IPermission } from '@/api/system/permission';

export interface IRole {
    id?: number
    name: string;
    code: string;
    permissions?: IPermission[]
    permission_ids?: number[]
}

// 创建角色
export const addApi = (data: IRole): IRespData<null> => request.post("/Role/Add", data)
// 编辑角色
export const editApi = (data: Required<IRole>): IRespData<null> => request.post("/Role/Edit", data)
// 删除角色
export const deleteApi = (id: number): IRespData<null> => request.post("/Role/Delete", { id })
// 根据 ID 获取角色
export const getApi = (id: number): IRespData<Required<IRole>> => request.get("/Role/Get", { params: { id } })
// 分页查询角色
export const listPageApi = (data: IPager): IRespData<IPageData<Required<IRole>>> => request.post("/Role/ListPage", data)
// 查询所有角色
export const listAllApi = (): IRespData<Required<IRole>[]> => request.get("/Role/ListAll")

File: app/models/view/system/Role.go
package v_system

type AddRoleReq struct {
	Name          string `json:"name" valid:"required~角色名称不能为空"`
	Code          string `json:"code" valid:"required~角色代码不能为空"`
	PermissionIDs []int  `json:"permission_ids"`
}

type EditRoleReq struct {
	ID int `json:"id" valid:"int~角色ID必须为整型数字,gt(0)~角色ID必须大于0"`
	AddRoleReq
}
